nncf/quantization/algorithms/hyperparameter_tuner/backend.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC
+from abc import abstractmethod
+from typing import Any, TypeVar
+
+TModel = TypeVar("TModel")
+
+
+class HyperparameterTunerAlgoBackend(ABC):
+    @staticmethod
+    @abstractmethod
+    def prepare_for_inference(model: TModel) -> Any:
```

Comment:
```suggestion
    def prepare_for_inference(model: TModel) -> TModel:
```
Why not `TModel`?

Reply:
Return type is different from input type.
input type for OV: `ov.Model`
return type for OV: `ov.CompiledModel`